lib/matplotlib/backends/_backend_tk.py

```diff
@@ -65,11 +90,32 @@ def blit(photoimage, aggimage, offsets, bbox=None):
         y1 = max(math.floor(y1), 0)
         y2 = min(math.ceil(y2), height)
         bboxptr = (x1, x2, y1, y2)
+        blank = False
     else:
-        photoimage.blank()
         bboxptr = (0, width, 0, height)
-    _tkagg.blit(
-        photoimage.tk.interpaddr(), str(photoimage), dataptr, offsets, bboxptr)
+        blank = True
+
+    # NOTE: _tkagg.blit is thread unsafe and will crash the process if called
+    # from a thread (GH#13293). Instead of blanking and blitting here,
+    # use tkapp.call to post a cross-thread event if this function is called
+    # from a non-Tcl thread.
+
+    # tkapp.call coerces all arguments to strings, so to avoid string parsing
+    # within _blit, pack up the arguments into a global data structure.
+    args = photoimage, dataptr, offsets, bboxptr, blank
+    # Need a unique key to avoid thread races.
+    # Again, make the key a string to avoid string parsing in _blit.
+    argsid = repr(id(args))
+    _blit_args[argsid] = args
+
+    global _blit_tcl_name
+    try:
+        photoimage.tk.call(_blit_tcl_name, argsid)
+    except tk.TclError:
+        # register _blit with code copied from tkinter.Misc._register
+        _blit_tcl_name = repr(id(_blit)) + _blit.__name__
```

Comment:
as above, could this just use a single global name defined at the toplevel?

Reply:
very yes